cmd/dlt/oidcconfig/cmd.go

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oidcconfig
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/briandowns/spinner"
+	"github.com/spf13/cobra"
+	"github.com/zgalor/weberr"
+
+	"github.com/openshift/rosa/pkg/arguments"
+	"github.com/openshift/rosa/pkg/aws"
+	awscb "github.com/openshift/rosa/pkg/aws/commandbuilder"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "oidc-config",
+	Aliases: []string{"oidconfig"},
+	Short:   "Delete OIDC Config",
+	Long:    "Cleans up OIDC config base on secret ARN.",
+	Example: `  # Delete OIDC config secret ARN that has been supplied
+	rosa delete oidc-config --oidc-private-key-secret-arn <oidc_private_key_secret_arn>`,
+	Hidden: true,
+	Run:    run,
+}
+
+const (
+	//nolint
+	OidcPrivateKeySecretArnFlag = "oidc-private-key-secret-arn"
+
+	prefixForPrivateKeySecret = "rosa-private-key-"
+)
+
+var args struct {
+	oidcPrivateKeySecretArn string
+	region                  string
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVar(
+		&args.oidcPrivateKeySecretArn,
+		OidcPrivateKeySecretArnFlag,
+		"",
+		"AWS Secrets Manager ARN for identification of config",
+	)
+
+	aws.AddModeFlag(Cmd)
+
+	interactive.AddFlag(flags)
+	confirm.AddFlag(flags)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithAWS()
+	defer r.Cleanup()
+
+	mode, err := aws.GetMode()
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
```

Comment:
should this be `%w`?

Reply:
Makes sense, but we need to check. Today we don't almost don't use `%w` so I think we should refactor alongside everything else at some other point

Reply:
I noticed that as well, we should be using this when possible for error wrapping. Maybe this can be tackled at another point in time when we can refactor every error message